test/e2e/network/service.go

```diff
@@ -1684,9 +1684,24 @@ var _ = common.SIGDescribe("Services", func() {
 		service.Spec.Type = v1.ServiceTypeNodePort
 
 		ginkgo.By("creating service " + serviceName + " with type NodePort in namespace " + ns)
-		service, err := t.CreateService(service)
-		framework.ExpectNoError(err, "failed to create service: %s in namespace: %s", serviceName, ns)
-
+		// Retry mechanism for picking a node port between 30000 and 30085
+		// to avoid conflicts
+		// https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3668-reserved-service-nodeport-range
```

Comment:
Use `e2eservice.NodePortRange`. (It's still hardcoded, but at least that way it's only hardcoded in one place.)

Also, the comment doesn't really explain what you're doing very well (and I initially thought you were fixing the bug in the wrong place).

"For the test to work, we need to make sure the Service's NodePort doesn't get reallocated to another Service after service is deleted. So we have to find an unused NodePort in the static allocation range."

Reply:
👍